src/group/errors.rs

```diff
@@ -0,0 +1,43 @@
+// maelstrom
+// Copyright (C) 2020 Raphael Robert
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+
+pub enum WelcomeError {
```

Comment:
So this should get into the spec as well as enums.
I have to look at these a little closer but have to make sure that the level of detail is not an issue. At first sight it looks like a lot of different error code.